client/client.go

```diff
@@ -371,6 +371,19 @@ func (d *dynamicWatcher) AddOrUpdateWatcher(watcher ObjectIdentifier, watchedObj
 	watchesAdded := []ObjectIdentifier{}
 
 	for _, watchedObject := range watchedObjects {
+		gvr, namespaced, err := d.gvrFromObjectIdentifier(watchedObject)
+		if err != nil {
+			klog.Errorf("Could not get the GVR for %s, error: %v", watchedObject, err)
+
+			encounteredErr = err
+
+			break
+		}
+
+		if !namespaced { // ignore namespaces set on cluster-scoped resources
+			watchedObject.Namespace = ""
+		}
```

Comment:
Should this return an error rather than wiping away the namespace?

Reply:
That would be a breaking change.